### PR TITLE
Make TTS queue a singleton

### PR DESCRIPTION
**Pull request by ken-mycroft**
## Description
Fixes issue where when remote tts fails and local gets executed they stomp on each other

## How to test
You would need a special mimic2_tts.py that periodically throws exceptions, otherwise not easy to test

## Contributor license agreement signed?
CLA [ yes ] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

**Review by forslund — APPROVED** [review]
Super nice, changing this to a singleton is a good way to handle this. Was looking into this the other week when I was debugging the weird zork-skill early bings when running on Mimic2. I was considering injecting the queue but this is a much less intrusive change.

Maybe this single Change could go directly into dev?

File: mycroft/tts/tts.py [hunk 1]
```diff
@@ -355,9 +355,6 @@ def execute(self, sentence, ident=None, listen=False):
         try:
             self._execute(sentence, ident, listen)
         except Exception:
-            # If an error occurs end the audio sequence through an empty entry
-            self.queue.put(EMPTY_PLAYBACK_QUEUE_TUPLE)
-            # Re-raise to allow the Exception to be handled externally as well.
             raise
```
Comment (forslund) [inline mycroft/tts/tts.py:358]: Maybe the try-except can be removed completely when the except just re-raises the exception.

**Comment by forslund** [comment]
One thing to note is that we might also need update the handling of the tts-cache. I think in this setup only one of the TTS's cache clearing methods will only be called...

**Comment by krisgesling** [comment]
I was hoping we could merge into both branches, however I've been testing this out on the standard dev channel as well and I think we must have some extra fixes to the mimic1 class in feature/mark-2. If you set your TTS module to mimic it starts playing all sentences in the queue at the same time.

**Comment by krisgesling** [comment]
Merging this now so we can get it onto tonights build but agree with the extra points raised so we should address those as a followup.

**Comment by forslund** [comment]
Ok interesting, I'll try that out and see if I can make this work for dev aswell.

**Comment by ken-mycroft** [comment]
> 

I was worried about this. There might be some mimic1/2 changes i missed. Is this still an issue?

**Comment by forslund** [comment]
I added a couple of things in the backport to dev PR #3055, it has commit for handling the cache clearing and some minor reorganization moving responsibility to the playback-thread. It also has some further fixes where the fallback could misbehave

**Pull request merged**